src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs

```diff
@@ -115,6 +116,17 @@ private static void WorkerThreadStart()
                         }
                     }
 
+                    // The thread cannot exit if it has IO pending, otherwise the IO may be canceled
+                    bool success =
+                        Interop.Kernel32.GetThreadIOPendingFlag(
```

Comment:
Isn't this code meant to be portable across OSes?

Reply:
Ah yes, will fix